### PR TITLE
Lumosviridi v20 kubernetes updates

### PR DESCRIPTION
**Pull request by LumosViridi**
Updates for v20+ and misc terraform bug fixes. Also refactored to use terraform variables instead of locals which helps with readability and ease of use for new users. 

Terraform validation is currently passing:
Screenshot 2024-07-21 at 13 18 37

Additionally added [terraform-docs](https://terraform-docs.io/) to generate a more helpful README for terraform specific configuration.

Raw K8s manifests were updated with changes for v20+ as well.

### CODE REVIEW

File: packages/twenty-docker/k8s/manifests/deployment-server.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ spec:
           value: 3000
         - name: SERVER_URL
           value: "https://crm.example.com:443"
+        - name: FRONT_BASE_URL
+          value: "https://crm.example.com:443"
+        - name: BACKEND_SERVER_URL
```
Comment (charlesBochet) [inline packages/twenty-docker/k8s/manifests/deployment-server.yaml:36]: actually, I am not sure this one is useful, it is not used in the codebase anymore

File: packages/twenty-docker/k8s/manifests/deployment-server.yaml [hunk 2]
```diff
@@ -39,6 +43,8 @@ spec:
           value: "true"
         - name: STORAGE_TYPE
           value: "local"
+        - name: "MESSAGE_QUEUE_TYPE"
+          value: "pg-boss"
```
Comment (charlesBochet) [inline packages/twenty-docker/k8s/manifests/deployment-server.yaml:47]: LGTM for now but we are aiming for bull-mq (this needs a redis to be setup) setup long term which is more robust

File: packages/twenty-docker/k8s/manifests/deployment-server.yaml [hunk 3]
```diff
@@ -76,7 +82,7 @@ spec:
         stdin: true
         tty: true
         volumeMounts:
-        - mountPath: /app/.local-storage
```
Comment (charlesBochet) [inline packages/twenty-docker/k8s/manifests/deployment-server.yaml:79]: we actually need both: .local-storage and docker-data

**Review by charlesBochet — APPROVED** [review]
I love it! Thank you @LumosViridi

**Review by FelixMalfait — APPROVED** [review]
Thank you!

**Pull request merged**